python/chemiscope/adapters.py

```diff
@@ -54,15 +73,47 @@ def _ase_to_json(frame):
 
 
 def _ase_atom_properties(frames):
+    # extract the set of common properties between all frames
+    all_names = set()
+    extra = set()
+    for name in frames[0].arrays.keys():
+        if name in IGNORED_ASE_ARRAYS:
+            continue
+        all_names.add(name)
+
+    for frame in frames[1:]:
+        for name in frame.arrays.keys():
+            if name in IGNORED_ASE_ARRAYS:
+                continue
+
+            if name not in all_names:
+                extra.add(name)
+
+        remove = []
+        for name in all_names:
+            if name not in frame.arrays.keys():
+                remove.append(name)
+
+        for name in remove:
+            all_names.remove(name)
+            extra.add(name)
+
+    if len(extra) != 0:
+        warnings.warn(
+            "the following atomic properties properties are only defined "
+            + f"for a subset of frames: {list(sorted(extra))}; they will be ignored"
+        )
```

Comment:
Design choice: omission of partial properties opposed to partial property lists?

I strongly prefer partial property lists, where we might throw a warning saying "xx property found for xx/xx frames". This will avoid a property being omitted across the board if it is missing from one frame.

Reply:
Well, chemiscope does not support partial property list for now =). There is the NaN hack, but it is a hack, and I would much rather leave this kind of decision to the user. This is already dealing with a very corner case, and the user gets a warning if something strange is going on, so they can act on it.

---- 

Overall, I would be in favor of **NOT** loading properties from ASE automatically, and instead have the user always provide what they want us to load. We could have an API like this:

```py
from chemiscope import create_input

data = create_input(
    frames = frames,
    properties = {
        "PCA": np.array(),  # automatically guess target
        "something else": {"target": "atom", "values": XXX},  # or allow to specify it
    },
    atom_properties_from_frames = ["name", "other name"],
    structure_properties_from_frames = ["something else"],
    meta={},  # as now
    cutoff=4.5,  # as now
)
```

This should be left for a future PR though.

Reply:
Okay, let's do it this way for now and revisit later